pkgs/tools/misc/coreutils/default.nix

```diff
@@ -10,27 +19,27 @@
 , singleBinary ? "symlinks" # you can also pass "shebangs" or false
 }:
 
-# Note: this package is used for bootstrapping fetchurl, and thus
-# cannot use fetchpatch! All mutable patches (generated by GitHub or
-# cgit) that are needed here should be included directly in Nixpkgs as
-# files.
+# Note: this package is used for bootstrapping fetchurl, and thus cannot use
+# fetchpatch! All mutable patches (generated by GitHub or cgit) that are needed
+# here should be included directly in Nixpkgs as files.
 
 assert aclSupport -> acl != null;
 assert selinuxSupport -> libselinux != null && libsepol != null;
 
-with lib;
-
+let
+  isNotCross = (stdenv.hostPlatform == stdenv.buildPlatform);
+in
 stdenv.mkDerivation (rec {
-  pname = "coreutils${optionalString (!minimal) "-full"}";
+  pname = "coreutils${lib.optionalString (!minimal) "-full"}";
   version = "9.1";
 
   src = fetchurl {
     url = "mirror://gnu/coreutils/coreutils-${version}.tar.xz";
-    sha256 = "sha256:08q4b0w7mwfxbqjs712l6wrwl2ijs7k50kssgbryg9wbsw8g98b1";
+    sha256 = "sha256-YaH0ENeLp+fzelpPUObRMgrKMzdUhKMlXt3xejhYBCM=";
```

Comment:
```suggestion
    hash = "sha256-YaH0ENeLp+fzelpPUObRMgrKMzdUhKMlXt3xejhYBCM=";
```

Reply:
By some strange reason it does not accept the `hash` argument...

Reply:
Has the boot fetchurl hash agility?